### PR TITLE
[MRG+1] EGI channel naming.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4076

### CODE REVIEW

File: mne/io/egi/egi.py [hunk 1]
```diff
@@ -103,7 +103,8 @@ def _combine_triggers(data, remapping=None):
 
 @verbose
 def read_raw_egi(input_fname, montage=None, eog=None, misc=None,
-                 include=None, exclude=None, preload=False, verbose=None):
+                 include=None, exclude=None, channel_naming='E%d',
+                 preload=False, verbose=None):
```
Comment (agramfort) [inline mne/io/egi/egi.py]: can you put it after preload to avoid API break?

**Comment by agramfort** [comment]
@espressofiend does it do the job for you?

File: mne/io/egi/egi.py [hunk 2]
```diff
@@ -154,6 +155,12 @@ def read_raw_egi(input_fname, montage=None, eog=None, misc=None,
 
         ..versionadded:: 0.11
 
+    channel_naming : str
+        Channel naming convention for the data channels. Defaults to 'E%d'
+        (resulting in channel names 'E1', 'E2', 'E3'...).
```
Comment (Eric89GXL) [inline mne/io/egi/egi.py]: ... the effective default prior to 0.14.0 was `'EEG %03d'`.

File: mne/io/egi/tests/test_egi.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def test_io_egi():
     include = ['TRSP', 'XXX1']
     with warnings.catch_warnings(record=True):  # preload=None
         raw = _test_raw_reader(read_raw_egi, input_fname=egi_fname,
-                               include=include)
+                               include=include, channel_naming='EEG %03d')
```
Comment (Eric89GXL) [inline mne/io/egi/tests/test_egi.py]: Do you need to make this change? `_test_raw_reader` shouldn't really impose channel naming requirements.

**Comment by Eric89GXL** [comment]
Otherwise LGTM

File: mne/io/egi/tests/test_egi.py (hunk 3 above)
Comment (jaeilepp) [inline mne/io/egi/tests/test_egi.py]: This is for the check 4 lines below. I can also change the check.

Comment (Eric89GXL) [inline mne/io/egi/tests/test_egi.py]: ahh okay, not necessary then

Comment (Eric89GXL) [inline mne/io/egi/tests/test_egi.py]: (and by that, I mean what you have now is fine as is)

**Comment by jaeilepp** [comment]
I just changed the check. Shouldn't make a difference. Ready for merge from my end.

**Comment by Eric89GXL** [comment]
Changes LGTM, +1 for merge

**Comment by espressofiend** [comment]
@agramfort yes it works! Thanks everyone!

**Comment by agramfort** [comment]
great !

**Comment by fraimondo** [comment]
Hey guys,

Sorry to bust this merge, but it introduced a bug when no stim channels are present.

It just needs  `event_codes = []` at mne/io/egi/egi.py:204

**Comment by agramfort** [comment]
can you open a PR?

**Comment by Eric89GXL** [comment]
[Done and merged](https://github.com/mne-tools/mne-python/pull/4212)

**Comment by agramfort** [comment]
ok ok I am too slow...

**Pull request merged**